docs/content/guides/dagster/local_to_prod_tutorial.mdx

```diff
@@ -0,0 +1,393 @@
+    assert 1 in out_ids
+    assert 3 not in out_ids
+```
+
+### Part Three: Production
+
+Now that our assets work locally and can be tested, let’s set up for the production environment. We want to store the assets in a production Snowflake database, so we need to update the configuration for the `snowflake_io_manager`
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v3.py startafter=start endbefore=end
+# repository.py
+
+@repository
+def repo():
+    resource_defs = {
+        "hn_client": hn_api_client,
+        "io_manager": snowflake_io_manager.configured(
+            {
+                "account": "abc1234.us-east-1",
+                "user": "dev@company.com",
+                "password": "company_super_secret_password",
+                "database": "PRODUCTION",
+                "schema": "HACKER_NEWS"
+            }
+        ),
+    }
+
+    return [*with_resources([items, comments, stories], resource_defs=resource_defs)]
+```
+
+Here we run into an issue. The next time a developer wants to work on the assets, they will have to remember to change the configuration for the `snowflake_io_manager`. This leaves room for a developer to accidentally override the production asset during local development. Instead we can set up the repository to attach resources to assets based on the environment
+
+```python file=/guides/dagster/local_to_prod_tutorial/repository/repository_v4.py startafter=start endbefore=end
+# repository.py
+
+@repository
+def repo():
+    resource_defs = {
+        "local": {
+            "hn_client": hn_api_client,
+            "io_manager": snowflake_io_manager.configured({
+                    "account": "abc1234.us-east-1",
+                    "user": "me@company.com",
+                    "password": "my_super_secret_password",
+                    "database": "SANDBOX",
+                    "schema": "JAMIE"
+            })
+        },
+        "production": {
+            "hn_client": hn_api_client,
+            "io_manager": snowflake_io_manager.configured({
+                    "account": "abc1234.us-east-1",
+                    "user": "dev@company.com",
+                    "password": "company_super_secret_password",
+                    "database": "PRODUCTION",
+                    "schema": "HACKER_NEWS"
+            })
+        }
+    }
+    deployment_name = os.environ.get("DAGSTER_DEPLOYMENT", "local")
+
+    return [
+        with_resources([items, comments, stories], resource_defs=resource_defs[deployment_name])
+    ]
+```
+
+Now we just set the environment variable `DAGSTER_DEPLOYMENT=production` in our deployment and the correct resources will be applied to the assets.
+
+<Note>
+
+If you are using \[Dagster Cloud]\(link to dagster cloud page explaining this?), this environment variable will be set for you. Additionally, if you are using \[Branch Deployments]\(link to branch deployments page explaining this?) `DAGSTER_DEPLOYMENT` will be automatically set to `VAR`.
```

Comment:
TODO - link to dagster cloud page. What page should this be?

TODO - link to branch deployments page